applications/GeoMechanicsApplication/custom_strategies/schemes/generalized_newmark_scheme.hpp

```diff
@@ -83,11 +125,52 @@ class GeneralizedNewmarkScheme
             (mTheta * this->GetDeltaTime());
     }
 
-private:
+    void UpdateVectorFirstTimeDerivative(Node& rNode) const
+    {
+        for (const auto& r_second_order_vector_variable :
+             this->GetSecondOrderVectorVariables())
+        {
+            if (!rNode.SolutionStepsDataHas(r_second_order_vector_variable.instance))
+                continue;
+
+            noalias(rNode.FastGetSolutionStepValue(
+                r_second_order_vector_variable.first_time_derivative, 0)) =
+                rNode.FastGetSolutionStepValue(
+                    r_second_order_vector_variable.first_time_derivative, 1) +
+                (1.0 - mGamma) * this->GetDeltaTime() *
+                    rNode.FastGetSolutionStepValue(
+                        r_second_order_vector_variable.second_time_derivative, 1) +
+                mGamma * this->GetDeltaTime() *
+                    rNode.FastGetSolutionStepValue(
+                        r_second_order_vector_variable.second_time_derivative, 0);
+        }
+    }
+
+    void UpdateVectorSecondTimeDerivative(Node& rNode) const
+    {
+        for (const auto& r_second_order_vector_variable :
+             this->GetSecondOrderVectorVariables())
+        {
+            if (!rNode.SolutionStepsDataHas(r_second_order_vector_variable.instance))
+                continue;
+
+            noalias(rNode.FastGetSolutionStepValue(
+                r_second_order_vector_variable.second_time_derivative, 0)) =
+                ((rNode.FastGetSolutionStepValue(r_second_order_vector_variable.instance, 0) -
+                  rNode.FastGetSolutionStepValue(r_second_order_vector_variable.instance, 1)) -
+                 this->GetDeltaTime() *
+                     rNode.FastGetSolutionStepValue(
+                         r_second_order_vector_variable.first_time_derivative, 1) -
+                 (0.5 - mBeta) * this->GetDeltaTime() * this->GetDeltaTime() *
+                     rNode.FastGetSolutionStepValue(
+                         r_second_order_vector_variable.second_time_derivative, 1)) /
+                (mBeta * this->GetDeltaTime() * this->GetDeltaTime());
+        }
+    }
+
     double mTheta = 1.0;
-    Variable<double> mVariable;
-    Variable<double> mDeltaTimeVariable;
-    Variable<double> mDeltaTimeVariableCoefficient;
+    double mBeta = 0.25;
+    double mGamma = 0.5;
```

Comment:
We may consider to use `std::optional<double>`s here, since not all of them are needed for all problems.

Reply:
Changed to optional